action.yml

```diff
@@ -57,10 +57,48 @@ inputs:
   actor:
     description: "The github actor responsible for the change"
     required: true
+  pipelines_token:
+    description: "The GitHub token for downloading the Gruntwork Pipelines binary"
+    required: false
+  pipelines_cli_version:
+    description: "The version of the Gruntwork Pipelines CLI to use"
+    required: false
+    default: "0.3.0"
+  pipelines_auth_role:
+    description: "The role to assume before running `pipelines auth presign`, which is used to authenticate the calling repo. Only necessary when pipelines auth role is provisioned and checked in infrastructure-pipelines repo."
+    required: false
+  pipelines_auth_region:
+    description: "The region in which the role to assume before running `pipelines auth presign`. Only necessary when pipelines auth role is provisioned and checked in infrastructure-pipelines repo."
+    required: false
+    default: "us-east-1"
 
 runs:
   using: "composite"
   steps:
+    - name: Download Pipelines CLI
+      if: ${{ inputs.pipelines_token != '' }}
+      uses: dsaltares/fetch-gh-release-asset@1.1.1
+      with:
+        repo: "gruntwork-io/pipelines-cli"
+        version: "tags/${{ inputs.pipelines_cli_version }}"
+        file: "pipelines_linux_amd64"
+        target: "/tmp/pipelines"
+        token: ${{ inputs.token }}
+
+    - name: Install Pipelines CLI
+      if: ${{ inputs.pipelines_token != '' }}
+      shell: bash
+      run: |
+        sudo mv /tmp/pipelines /usr/local/bin/pipelines
+        sudo chmod +x /usr/local/bin/pipelines
+
+    - name: Authenticate with AWS
+      if: ${{ inputs.pipelines_auth_role != '' }}
+      uses: aws-actions/configure-aws-credentials@v4
+      with:
+        role-to-assume: arn:aws:iam::${{ inputs.account_id }}:role/${{ inputs.pipelines_auth_role }}
+        aws-region: ${{ inputs.pipelines_auth_region }}
```

Comment:
Do we really want to assume the role? Or do we want to merely presign a request for assuming the role?

Reply:
We want to really assume the role, but not really send the `GetCallerIdentity` API request.

We don't have any AWS credentials in the GitHub Actions runner until we assume the role, as we only have the token to make the OIDC role assumption with the `AssumeRole` API call. We use those credentials to presign the `GetCallerIdentity` API request, then send that down to another work flow which issues it.

Reply:
Oh, yea, that makes sense, thanks!